watertap/unit_models/tests/unit_test_harness.py

```diff
@@ -168,101 +168,6 @@ def test_unit_solutions(self, frame_unit):
-            #     var = getattr(blk, key[0])[
-            #         key[1]
-            #     ]  # m.fs.feed_side.properties_in.flow_mass_phase_comp[Liq, NaCl]
-            #     # relative tolerance doesn't mean anything for 0-valued things
-            #     if val == 0:
-            #         if not pytest.approx(val, abs=1.0e-08) == value(var):
-            #             raise UnitValueError(
-            #                 "Variable {v_name} is expected to have a value of {val} +/- 1.0e-08, but it "
-            #                 "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-            #                 "that sets up the UnitRegressionTest".format(
-            #                     v_name=key[0], ind=key[1], val=val, val_t=value(var)
-            #                 )
-            #             )
-            #     elif not pytest.approx(val, rel=1e-3) == value(var):
-            #         raise UnitValueError(
-            #             "Variable {v_name} is expected to have a value of {val} +/- 0.1%, but it "
-            #             "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-            #             "that sets up the UnitRegressionTest".format(
-            #                 v_name=key[0], ind=key[1], val=val, val_t=value(var)
-            #             )
-            #         )
-
-        # for (port, stateblock, v_name, ind), val in solution.items():
-        #     cv_blk = getattr(blk, port)  # m.fs.feed_side
-        #     stream = getattr(cv_blk, stateblock)  # m.fs.feed_side.properties_in
-        #     var = getattr(stream[0], v_name)[ind]  # m.fs.feed_side.properties_in.flow_mass_phase_comp[Liq, NaCl]
-        #     # relative tolerance doesn't mean anything for 0-valued things
-        #     if val == 0:
-        #         if not pytest.approx(val, abs=1.0e-08) == value(var):
-        #             raise UnitValueError(
-        #                 "Variable {v_name} is expected to have a value of {val} +/- 1.0e-08, but it "
-        #                 "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-        #                 "that sets up the UnitRegressionTest".format(
-        #                     v_name=v_name, ind=ind, val=val, val_t=value(var)
-        #                 )
-        #             )
-        #     elif not pytest.approx(val, rel=1e-3) == value(var):
-        #         raise UnitValueError(
-        #             "Variable {v_name} is expected to have a value of {val} +/- 0.1%, but it "
-        #             "has a value of {val_t}. \nUpdate unit_solution in the configure function "
-        #             "that sets up the UnitRegressionTest".format(
-        #                 v_name=v_name, ind=ind, val=val, val_t=value(var)
-        #             )
-        #         )
+            pyo_obj = self.model.find_component(key)
+            assert pytest.approx(value(pyo_obj), rel=1e-5) == val
```

Comment:
I think we should think through a slick way to supply relative and absolute tolerances which are dependent on the value, while not overly burdening the test harness implementer if it doesn't matter for their model.

In particular, we have a lot of values close to `0` where a relative tolerance of `1e-5` is too tight.

Reply:
Off the top of my head:

- `self.unit_solutions` values can be either:
  - A single number, in which case a default (per-class?) tolerance is used
  - A tuple `(val, tol)` where the value is specified together with a custom tolerance

The added complication for the `UnitTestHarness` code is... tolerable (pun clearly intended), e.g.:

```py
for key, val in solutions.items():
    expected, tol = val if isinstance(val, tuple) else val, self.default_tol
    assert pytest.approx(value(pyo_obj), tol=tol) == expected
```

Reply:
is `pyo_obj` still `self.model.find_component(key)` in this case? Assuming so, I'm still seeing the same AssertionError where `pyo_obj` is `None`.

Also, the first tol in `assert pytest.approx(value(pyo_obj), tol=tol) == expected` has to be either `rel` or `abs` I think.